beetsplug/parentwork.py

```diff
@@ -0,0 +1,165 @@
+# -*- coding: utf-8 -*-
+# This file is part of beets.
+# Copyright 2017, Dorian Soergel.
+#
+# Permission is hereby granted, free of charge, to any person obtaining
+# a copy of this software and associated documentation files (the
+# "Software"), to deal in the Software without restriction, including
+# without limitation the rights to use, copy, modify, merge, publish,
+# distribute, sublicense, and/or sell copies of the Software, and to
+# permit persons to whom the Software is furnished to do so, subject to
+# the following conditions:
+#
+# The above copyright notice and this permission notice shall be
+# included in all copies or substantial portions of the Software.
+
+"""Gets work title, disambiguation, parent work and its disambiguation,
+composer, composer sort name and performers
+"""
+
+from __future__ import division, absolute_import, print_function
+
+from beets import ui
+from beets.plugins import BeetsPlugin
+from beets.dbcore import types
+
+import musicbrainzngs
+
+def find_parentwork(work_id):
+    """This function finds the parentwork of a work given its id. """
+    work_info = musicbrainzngs.get_work_by_id(work_id,
+        includes=["work-rels", "artist-rels"])
+    partof = True
+    # The works a given work is related to are listed in
+    # work_info['work']['work-relation-list']. The relations can be
+    # diverse: arrangement of, later version of, part of etc. The
+    # father work (i. e. the work the work is part of) is given by the
+    # relationship 'type' to be 'part' and the relationship
+    # 'direction' to be 'backwards'. First I assume the work doesn't
+    # have a father work (i. e. it is its own parentwork), but if in
+    # the works it is related to there is a work which is his father
+    # work, then I assume the father work is the parent work and try
+    # the same with it.
+    while partof:
+        partof = False
+        if 'work-relation-list' in work_info['work']:
+            for work_father in work_info['work']['work-relation-list']:
+                if work_father['type'] == 'parts' and work_father.get(
+                    'direction') == 'backward':
+                    father_id = work_father['work']['id']
+                    partof = True
+                    work_info = musicbrainzngs.get_work_by_id(
+                        father_id,includes = ["work-rels","artist-rels"])
+    return work_info
+
+def get_info(work_info,parent_composer,parent_composer_sort,parent_work,
+    parent_work_disambig,work_ids,composer_ids):
+    """Given the parentwork info dict, this function updates the 
+    parent composer etc"""
```

Comment:
In docstrings, it's a good idea to mention which parameters are "input" parameters and which are "output" parameters. If it's not obvious from the name, it's in general useful to describe what the parameters mean.